### PR TITLE
RADA Email Server, Validation Email Failure Checking.

### PR DESCRIPTION
**Pull request by nickwxo**
This pull request implements the RADA email server in place of Sendgrid and provides the user with a visual notification if the email has not been sent. If an activation email cannot be sent the user account is deleted immediately (this is to allow them to create it again with the same information later without duplicate username or email address problems).

Additional small tweak : 

* The address field should never have been mandatory on the User model. This presented a problem during my testing so I fixed it.

Fixes Issue #59

### CODE REVIEW

**Comment by nickwxo** [comment]
screen shot 2016-12-04 at 7 49 53 pm

File: models/db.js [hunk 1]
```diff
@@ -40,7 +40,7 @@ var UserSchema = new Schema({
     us_contact: {type: String, required: true},
     us_user_creation_date: {type: Date, default: Date.now()},
     us_state: {type: String, default: 'pending'},
-    us_address: {type: String, required: true},
```
Comment (matjames007) [inline models/db.js:43]: My memory fails me... but isn't the address field mandatory for users of the platform? @CamiMonte @nick22891 @paradoxxjm ?

**Review by matjames007 — APPROVED** [review]
All changes look functionally safe and can proceed.  Question asked within the code.  WIll not merge, but will approve changes.

File: models/db.js (hunk 1 above)
Comment (nickwxo) [reply models/db.js:43]: @matjames007 I just asked @CamiMonte and @paradoxxjm and they both said no.

**Pull request merged**